pkg/net/host_test.go

```diff
@@ -137,6 +137,9 @@ func TestHostUnmarshalJSON(t *testing.T) {
 		{[]byte(`"12play"`), &Host{"12play", 0, false}, false},
 		{[]byte(`"play-minio-io"`), &Host{"play-minio-io", 0, false}, false},
 		{[]byte(`"play--minio.io"`), &Host{"play--minio.io", 0, false}, false},
+		{[]byte(`"[fe80::8097:76eb:b397:e067%wlp2s0]"`), &Host{"fe80::8097:76eb:b397:e067%wlp2s0", 0, false}, false},
+		{[]byte(`"[fe80::8097:76eb:b397:e067]:9000"`), &Host{"fe80::8097:76eb:b397:e067", 9000, true}, false},
+		{[]byte(`"fe80::8097:76eb:b397:e067%wlp2s0"`), nil, true},
```

Comment:
Add  all negative test cases.

Reply:
With `%` it becomes invalid, will add some testcases for this as well.

Reply:
Yes, it becomes invalid if it is not enclosed by square brackets. we get `too many colons` error